### PR TITLE
[SPARK-41875][CONNECT][PYTHON] Add test cases for `Dataset.to()`

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?

1. This PR let the parameter of `Dataset.to()` the same as pyspark.

2. The connect's `Dataset.to()` lost some test cases.
This PR adds these test cases that refer https://github.com/apache/spark/blob/89666d44a39c48df841a0102ff6f54eaeb4c6140/python/pyspark/sql/tests/test_dataframe.py#L1464

### Why are the changes needed?
This PR adds these test cases for connect's `Dataset.to()`.

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
New test cases.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Comment by beliefer** [comment]
ping @zhengruifeng @HyukjinKwon

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 1]
```diff
@@ -644,6 +644,15 @@ def test_schema(self):
     def test_to(self):
         # SPARK-41464: test DataFrame.to()
 
+        cdf = self.connect.read.table(self.tbl_name)
+        df = self.spark.read.table(self.tbl_name)
+
+        def compare_to_schema():
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:650]: Can we explicitly pass `df` so we know what is being compared? I would also name it like `assert_eq_schema`

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 2]
```diff
@@ -652,11 +661,17 @@ def test_to(self):
...
         )
 
-        cdf = self.connect.read.table(self.tbl_name).to(schema)
-        df = self.spark.read.table(self.tbl_name).to(schema)
+        compare_to_schema()
+
+        # Change schema with struct
+        schema2 = StructType([StructField("struct", schema, False)])
+        from pyspark.sql import functions as SF
+        from pyspark.sql.connect import functions as CF
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:669]: Let's import them on the top

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 1 above)
Comment (beliefer) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:650]: `assert_eq_schema` looks is a better name.

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 2 above)
Comment (beliefer) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:669]: OK

**Review by amaliujia — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_basic.py [hunk 3]
```diff
@@ -644,6 +645,15 @@ def test_schema(self):
     def test_to(self):
         # SPARK-41464: test DataFrame.to()
 
+        cdf = self.connect.read.table(self.tbl_name)
```
Comment (zhengruifeng) [inline python/pyspark/sql/tests/connect/test_connect_basic.py:648]: what about simply:
```
cdf = ...
sdf = ...
for schema in [ schema1, schema2, schema3 ]:
            cdf_to = cdf.to(schema)
            sdf_to = sdf.to(schema)
            self.assertEqual(cdf_to.schema, sdf_to.schema)
            self.assert_eq(cdf_to.toPandas(), sdf_to.toPandas())
```

I prefer not adding extra comprasion function

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/tests/connect/test_connect_basic.py (hunk 3 above)
Comment (beliefer) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:648]: `cdf` and `sdf` changed.

Comment (zhengruifeng) [reply python/pyspark/sql/tests/connect/test_connect_basic.py:648]: ok

**Comment by zhengruifeng** [comment]
merged into master, thanks @beliefer

**Comment by beliefer** [comment]
@zhengruifeng @HyukjinKwon @amaliujia Thank you for all!

**Pull request closed without merging**